vm/systemSmartContracts/sovereignEEI.go

```diff
@@ -0,0 +1,34 @@
+package systemSmartContracts
+
+import (
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/vm"
+)
+
+type sovereignVMContext struct {
+	vm.ContextHandler
+}
+
+// NewSovereignVMContext creates a new sovereign vm context
+func NewSovereignVMContext(vmContext vm.ContextHandler) (*sovereignVMContext, error) {
```

Comment:
This is NewSystemVMEEI - SovereignVMContext sounds bad.

It is ok that you pushed this context handler, but it needs a new name. It is the execution environmnent interface for the SystemVM.